### PR TITLE
Fix binning issues

### PR DESCRIPTION
**Pull request by nanchenchen**
This is for #145

### CODE REVIEW

File: msgvis/static/SparQs/charts.js [hunk 1]
```diff
@@ -449,16 +449,55 @@
...
+                    var next_value = self.primaryValueLabel.get_next_value(primaryValue);
+                    if ( typeof(next_value) !== "undefined" ){
+                        range.max_time = next_value;
+                    }
+                    primaryValue = range;
+                }
+                else{
+                    primaryValue = {'value': primaryValue };
+                }
+                values.push(primaryValue);
```
Comment (michaelbrooks) [inline msgvis/static/SparQs/charts.js]: Let's just move `var values = [primaryValue];` down to here.

**Comment by michaelbrooks** [comment]
Oh, I see you already merged this. I was going to suggest adding a few comments to explain the new blocks of code on charts.js, if you have a moment.

**Comment by nanchenchen** [comment]
I can do it later. As for the var values, I thought it should be var values
= []; now?
That does not make sense to me to be put to be above only one of them?

2015-04-30 10:26 GMT-07:00 Michael Brooks <notifications@github.com>:

> Oh, I see you already merged this. I was going to suggest adding a few
> comments to explain the new blocks of code on charts.js, if you have a
> moment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hds-lab/textvisdrg/pull/168#issuecomment-97888414>.
>



-- 
Nan-Chen Chen
nanchen.chen@gmail.com
PhD Student
Department of Human Centered Design & Engineering
University of Washington
Seattle, WA 98195
http://nanchen.csie.org/

**Comment by michaelbrooks** [comment]
Yeah, but then we just always push primaryValue onto it anyway, if I understand correctly. The code you added is just setting what primaryValue is before adding it to the array. But I guess since we then add a secondary value, it makes as much sense to just push both of them in the same way.

**Comment by nanchenchen** [comment]
That's a totally different mental model from mine. lol
I would switch the following two lines to make it more organized in the way
that makes more sense.

var primaryValue = self.primaryValueLabel.inverse(data.x);
var values = [];

2015-04-30 10:32 GMT-07:00 Michael Brooks <notifications@github.com>:

> Yeah, but then we just always push primaryValue onto it anyway, if I
> understand correctly. The code you added is just setting what primaryValue
> is before adding it to the array. But I guess since we then add a secondary
> value, it makes as much sense to just push both of them in the same way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hds-lab/textvisdrg/pull/168#issuecomment-97890425>.
>



-- 
Nan-Chen Chen
nanchen.chen@gmail.com
PhD Student
Department of Human Centered Design & Engineering
University of Washington
Seattle, WA 98195
http://nanchen.csie.org/

**Comment by michaelbrooks** [comment]
Yeah I think your way makes more sense after looking at it more.

**Pull request merged**